### ISSUE TITLE
Setuptools>=21 Compatibility

### ISSUE THREAD

**Issue by mlococo** [pantsbuild/pex#301]
versions.py specifies that pex is compatible with setuptools versions up to 20.10.x: https://github.com/pantsbuild/pex/blob/fa3bfc20155c015fc991f0dac0f64d23ebaf6f35/pex/version.py#L6

Experimentally, my own builds confirm this. I have a project structured as:

```
my-package
├── [  14]  build.properties
├── [  37]  MANIFEST.in
├── [   4]  my-package
│   ├── [155K]  core.py
│   └── [ 347]  __init__.py
├── [3.4K]  README.md
├── [   0]  requirements.txt
├── [3.4K]  setup.py
```

I built the project using pex as such and for setuptools 20.10.1 it builds as expected, but for newer versions of setuptools the package is unable to find itself:

```
$ pex --disable-cache -r my-package/requirements.txt -e my-package:main -o my-package.pex my-package
Could not satisfy all requirements for my-package:
    my-package
```

Is it known work that needs to happen to get pex compatible with newer setuptools?

**Comment by kwlzn** [comment]
I believe the pinning there is arbitrary - e.g. that was probably the latest or next setuptools version the last time that was arbitrarily bumped.

so assuming there's confirmed breakage in later versions, I doubt there's a good handle on known work to get to newer setuptools compatibility.

contributions are welcome tho if you have time to dig into this.

**Comment by mlococo** [comment]
Thanks for the historical context. I'll chat with folks on my end about whether we want to pin setuptools for our own builds, or if we want to dig into the resolver and chase down what needs updating. If I'm able to carve out time for the latter, I'll certainly submit a PR with the results. If not, this report can at least serve as a heads-up to users and a place to collect confirmation (in case my issue is not generally reproduceable).

**Comment by kwlzn** [comment]
fixed in #323
